cluster-api/src/main/java/io/aiven/klaw/clusterapi/services/ApacheKafkaAclService.java

```diff
@@ -0,0 +1,490 @@
+import org.apache.kafka.common.acl.AclBindingFilter;
+import org.apache.kafka.common.acl.AclOperation;
+import org.apache.kafka.common.acl.AclPermissionType;
+import org.apache.kafka.common.resource.PatternType;
+import org.apache.kafka.common.resource.ResourcePattern;
+import org.apache.kafka.common.resource.ResourcePatternFilter;
+import org.apache.kafka.common.resource.ResourceType;
+import org.springframework.core.env.Environment;
+import org.springframework.stereotype.Service;
+
+@Service
+@Slf4j
+public class ApacheKafkaAclService {
+
+  private static final long TIME_OUT_SECS_FOR_ACLS = 5;
+
+  private Environment env;
+
+  private ClusterApiUtils clusterApiUtils;
+
+  public ApacheKafkaAclService(ClusterApiUtils clusterApiUtils) {
+    this.clusterApiUtils = clusterApiUtils;
+  }
+
+  public ApacheKafkaAclService() {}
+
+  public ApacheKafkaAclService(Environment env) {
+    this.env = env;
+  }
+
+  public synchronized Set<Map<String, String>> loadAcls(
+      String environment, KafkaSupportedProtocol protocol, String clusterName) throws Exception {
+    log.info("loadAcls {} {}", environment, protocol);
+    Set<Map<String, String>> acls = new HashSet<>();
+
+    AdminClient client = clusterApiUtils.getAdminClient(environment, protocol, clusterName);
+    if (client == null) {
+      throw new Exception("Cannot connect to cluster.");
+    }
+
+    try {
+      AclBindingFilter aclBindingFilter = AclBindingFilter.ANY;
+      DescribeAclsResult aclsResult = client.describeAcls(aclBindingFilter);
+
+      aclsResult
+          .values()
+          .get(TIME_OUT_SECS_FOR_ACLS, TimeUnit.SECONDS)
+          .forEach(
+              aclBinding -> {
+                filterAndUpdateAclBindings(acls, aclBinding);
+              });
+    } catch (Exception e) {
+      log.error("Error " + e.getMessage());
+    }
+
+    return acls;
+  }
+
+  private static void filterAndUpdateAclBindings(
+      Set<Map<String, String>> acls, AclBinding aclBinding) {
+    if (aclBinding.pattern().patternType().name().equals(AclPatternType.LITERAL.value)) {
+      Map<String, String> aclbindingMap = new HashMap<>();
+      aclbindingMap.put("host", aclBinding.entry().host());
+      aclbindingMap.put("principle", aclBinding.entry().principal());
+      aclbindingMap.put("operation", aclBinding.entry().operation().toString());
+      aclbindingMap.put("permissionType", aclBinding.entry().permissionType().toString());
+      aclbindingMap.put("resourceType", aclBinding.pattern().resourceType().toString());
+      aclbindingMap.put("resourceName", aclBinding.pattern().name());
+
+      if (!aclBinding.pattern().resourceType().toString().equals("CLUSTER")) {
+        if (aclBinding.entry().operation().toString().equals("WRITE")
+            || aclBinding.entry().operation().toString().equals("READ")) {
+          acls.add(aclbindingMap);
+        }
+      }
+    }
+  }
+
+  public synchronized String updateProducerAcl(ClusterAclRequest clusterAclRequest) {
```

Comment:
May be i don't understand the internals however from my point of view synchronized could help only when there is only one jvm instance working with db...
In case there is more than one then it will not help...
Does it mean that only one instance of jvm is allowed to run?

Reply:
Acls are applied on kafka topics. There could be requests being concurrently executed on the same topic for create and delete, and at any point of time, allowing only one instance of jvm thread to run, to avoid any discrepanices

Reply:
in case of one instance of jvm it's clear.
My point is: is it possible to have several instances of Klaw connecting to the same Kafka?
If yes then `synchronized` will not help here

Reply:
It is possible that several klaw instances connect to same kafka. yea, and then synchronized would not work. got it. Will look into this problem in a different pr. thanks.